java/src/main/java/com/powsybl/python/PyPowsyblApiLib.java

```diff
@@ -291,6 +291,35 @@ public static void dumpNetwork(IsolateThread thread, ObjectHandle networkHandle,
         });
     }
 
+    @CEntryPoint(name = "dumpNetworkToString")
+    public static CCharPointer dumpNetworkToString(IsolateThread thread, ObjectHandle networkHandle, CCharPointer format,
+                                                   CCharPointerPointer parameterNamesPtrPtr, int parameterNamesCount,
+                                                   CCharPointerPointer parameterValuesPtrPtr, int parameterValuesCount,
+                                                   ExceptionHandlerPointer exceptionHandlerPtr) {
+        return doCatch(exceptionHandlerPtr, () -> {
+            Network network = ObjectHandles.getGlobal().get(networkHandle);
+            String formatStr = CTypeUtil.toString(format);
+            Properties parameters = createParameters(parameterNamesPtrPtr, parameterNamesCount, parameterValuesPtrPtr, parameterValuesCount);
+            MemDataSource dataSource = new MemDataSource();
+            var exporter = Exporters.getExporter(formatStr);
+            exporter.export(network, parameters, dataSource);
+            try {
+                var exts = dataSource.listNames(".*?");
+                if (exts.size() != 1) {
+                    throw new PowsyblException("Datasource file error");
```

Comment:
This is not really an error but a non handled case. We probably can do it later (not easy?) we should add a FIXME to explain it.

Reply:
et also a clear exception message that explain that will only support string export for single file format